subgraphs/rocket-pool/src/entities/token.ts

```diff
@@ -0,0 +1,153 @@
+  ONE_ETH_IN_WEI,
+  BIGDECIMAL_ONE,
+} from "../utils/constants";
+
+export function getOrCreateToken(
+  tokenAddress: Address,
+  blockNumber: BigInt
+): Token {
+  const tokenId = tokenAddress.toHexString();
+  let token = Token.load(tokenId);
+
+  if (!token) {
+    token = new Token(tokenId);
+
+    if (tokenAddress == Address.fromString(ETH_ADDRESS)) {
+      token.name = ETH_NAME;
+      token.symbol = ETH_SYMBOL;
+      token.decimals = 18;
+    } else {
+      token.name = fetchTokenName(tokenAddress);
+      token.symbol = fetchTokenSymbol(tokenAddress);
+      token.decimals = fetchTokenDecimals(tokenAddress) as i32;
+    }
+  }
+
+  // Optional lastPriceUSD and lastPriceBlockNumber, but used in financialMetrics
+  // if (tokenAddress.equals(Address.fromString(RETH_ADDRESS))) {
+  //   let rethpriceUSD: BigDecimal | null = null;
+  //   rethpriceUSD = getRETHpriceUSD(
+  //     Address.fromString(RETH_ADDRESS),
+  //     blockNumber
+  //   );
+  //   log.warning("[getOrCreateToken]rethPriceUSD call result: {} ", [
+  //     rethpriceUSD.toString(),
+  //   ]);
+
+  //   token.lastPriceUSD = rethpriceUSD;
+  //   token.lastPriceBlockNumber = blockNumber;
+  //   token.save();
+
+  //   return token;
+  // }
+  if (
+    token.lastPriceBlockNumber &&
+    token.lastPriceBlockNumber!.equals(blockNumber)
+  ) {
+    token.save();
+    return token;
+  }
+  let price = getUsdPricePerToken(tokenAddress);
+
+  if (price.reverted) {
+    token.lastPriceUSD = BIGDECIMAL_ZERO;
+  } else {
+    token.lastPriceUSD = price.usdPrice.div(price.decimalsBaseTen);
+  }
+
+  token.lastPriceBlockNumber = blockNumber;
+  token.save();
+
+  return token;
+}
+
+function fetchTokenName(tokenAddress: Address): string {
+  const tokenContract = ERC20.bind(tokenAddress);
+  const call = tokenContract.try_name();
+  if (call.reverted) {
+    return tokenAddress.toHexString();
+  } else {
+    return call.value;
+  }
+}
+
+function fetchTokenSymbol(tokenAddress: Address): string {
+  const tokenContract = ERC20.bind(tokenAddress);
+  const call = tokenContract.try_symbol();
+  if (call.reverted) {
+    return " ";
+  } else {
+    return call.value;
+  }
+}
+
+function fetchTokenDecimals(tokenAddress: Address): number {
+  const tokenContract = ERC20.bind(tokenAddress);
+  const call = tokenContract.try_decimals();
+  if (call.reverted) {
+    return 0;
+  } else {
+    return call.value;
+  }
+}
+
+function getRETHpriceUSD(
+  tokenAddress: Address,
+  blockNumber: BigInt
+): BigDecimal {
+  const RETHContract = RETH.bind(tokenAddress);
+
+  const rethCall = RETHContract.try_getExchangeRate();
+  let RETHratio = BIGDECIMAL_ZERO;
+  let RETHpriceUSD = BIGDECIMAL_ZERO;
+  if (rethCall.reverted) {
+    // log.error("[getRETHpriceUSD] rethCall Reverted", []);
+  } else {
+    // log.error("[getRETHpriceUSD] rethCall resolved to : {}", [
+    //   rethCall.value.toString(),
+    // ]);
+    RETHratio = bigIntToBigDecimal(rethCall.value.div(ONE_ETH_IN_WEI));
```

Comment:
I think this is wrong 🤔 

`getExchangeRate` returns the RETH price denominated in ETH, as a BigInt. If you divide by `ONE_ETH_IN_WEI` and then call `bigIntToBigDecimal` will divide it again by 10^18. Shouldn't it be just `bigIntToBigDecimal(value)`?

<img width="594" alt="Screenshot 2022-10-20 at 16 11 57" src="https://user-images.githubusercontent.com/17258946/196972642-f39349eb-64cd-4ef2-93b0-1732f5e7ca34.png">

Reply:
This function is unused.